README.md

```diff
@@ -1,5 +1,114 @@
+```typescript
+export interface IExportManager {
+  createExportTask: (data: CreateExportTaskRequest<TaskParameters>) => Promise<CrateExportTaskResponse>;
+  getEstimatedSize: (params?: GetEstimatedSizeParams) => Promise<number>;
+  getFileSize: (catalogRecordID: string, params: GetFileSizeParams) => Promise<number>;
+}
+```
+most of Export-Interfaces functions uses **[@turf/turf](https://www.npmjs.com/package/@turf/turf)** npm package for geographic data.
+
+---
+
+2. import & implement
+
+```typescript
+import { IExportManager } from '@map-colonies/export-interfaces';
+
+export class ExportManager implements IExportManager {
+    ...
+}
+```
+
+3. implements IExportManager functions
+
+- **createExportTask** example:
+```typescript
+export class ExportManager implements IExportManager {
+  public constructor(@inject(SERVICES.LOGGER) private readonly logger: Logger) {}
+
+  public async createExportTask(data: CreateExportTaskRequest<TaskParameters>): Promise<CreateExportTaskResponse> {
+    this.logger.info({ msg: 'log message' });
+    ...
+    ...
+    const geometry1: Polygon = {
+      coordinates: [
+        [
+          [38.12777806969146, 34.88509668647414],
+          [37.79098758535008, 34.88509668647414],
+          [37.79098758535008, 34.557014824377575],
+          [38.12777806969146, 34.557014824377575],
+          [38.12777806969146, 34.88509668647414],
+        ],
+      ],
+      type: 'Polygon',
+    };
+
+    const geometry2: MultiPolygon = {
+      coordinates: [
+        [
+          [30.10241695788926, 27.097838272461956],
+          [29.991351500386685, 27.0731170944439],
+          [29.991351500386685, 26.949429475478382],
+          [30.08853377570128, 26.776038590550044],
+          [30.310664690705238, 26.98655005029447],
+          [30.10241695788926, 27.097838272461956],
+        ],
+      ],
+      type: 'Polygon',
+    };
+
+    const taskGeometries: TaskGeometry[] = [{ geometry: geometry1, metadata: { maxRes: 0.732 } }, { geometry2 }]; // ⇨ 'metadata' is optional here
+    
+    const createExportTaskResponse: CreateExportTaskResponse = {
+      jobId: '4b3d6bcd-bbfd-4b2d-9x5d-ab8dfbdd4be4', // uuid format
+      geometries: taskGeometries,
+    }
+    ...
+
+    return createExportTaskResponse; 
+  }
+}
+```
+
+---
+
+- **getEstimatedSize** ⇨ returns the ***estimated*** exported **data** size
+- **getFileSize** ⇨ returns the ***actuall*** exported **data** size
+---
+
+### Event Message
+---
+Export-Interfaces exposes different event message types for each events 
+[Export-Management](https://github.com/MapColonies/export-management) usage (expects to receive those message objects in order to handle them while task event has occurred)
+
+### Task Events: 
++ TaskStartedMessage
++ TaskCompletedMessage
++ TaskFailedMessage
++ TaskAbortedMessage 
++ TaskExpiredMessage 
++ TaskPausedMessage 
+
+The event message types are intended to be used by the publishers and Export-Management as consumer
```

Comment:
I disagree, the protocol between the client to the export manager is via the following events.

The inner comms shouldn't be the same.